data-raw/england-aande.R

```diff
@@ -0,0 +1,91 @@
+# ---- Load libs ----
+library(tidyverse)
+library(readxl)
+library(devtools)
+library(httr2)
+library(lubridate)
+
+# ---- Load internal sysdata.rda file with URL's ----
+load_all(".")
+
+# ---- Function to download and clean ----
+scrape_data <- function(id, sheet, range, date, col_index) {
+  # get url from query_urls table
+  query_url <-
+    query_urls |>
+    filter(id == {{ id }}) |>
+    pull(query)
+
+  # download xls to as a temp file
+  download <- tempfile(fileext = ".xls")
+
+  request(query_url) |>
+    req_perform(download)
+
+  # read the xls file
+  raw <-
+    read_excel(
+      download,
+      sheet = sheet,
+      range = range
+    )
+
+  # clean the df
+  data <- raw |>
+    select(all_of(col_index)) |> # select relevant columns
+    rename(
+      code = 1, 
+      total_attendances = 2, 
+      attendances_over_4hours = 3, 
+      total_emergency_admissions = 4, 
+      emergency_admissions_over_4hours = 5
+    ) |> # rename columns
+    slice(-1) |> # remove first row
+    filter(!is.na(code)) |> # remove NAs
+    mutate_at(2:5, as.numeric) |> # change cols datatype to numeric
+    mutate(
+      pct_attendance_over_4hours = round(attendances_over_4hours / total_attendances, 2),
+      pct_emergency_admissions_over_4hours = round(emergency_admissions_over_4hours / total_emergency_admissions, 2),
+      date = date
+    ) # add percentage and date columns
+
+  return(data)
+}
+
+# ---- ICB level data ----
+# - Iterate over all data sets and return as a dataframe
+
+# Generate a dataframe with function arguments
+icb_df <-
+  tibble(
+    id = query_urls[which(query_urls$id == "nhs_accident_emergency_april_22"):(which(query_urls$id == "nhs_accident_emergency_april_22") + 12), ] |> pull(id),
```

Comment:
Considering you have to filter to specific dates, I think a cleaner way to write this code is:

```r
query_urls |> filter(between(id, "nhs_accident_emergency_april_22", "nhs_accident_emergency_march_23")) |> pull(id)
```

I think it makes the date ranges which are being filtered to more explicit.